queue/priority_queue.go

```diff
@@ -42,50 +39,66 @@ type Item interface {
 
 type priorityItems []Item
 
+func (items *priorityItems) swap(i, j int) {
+	(*items)[i], (*items)[j] = (*items)[j], (*items)[i]
+}
+
+func (items *priorityItems) pop() Item {
```

Comment:
Pointer receiver methods shouldn't be used in the case of slices. A slice is already a pointer. Instead, use the "by value" syntax:

```go
func (items priorityItems) pop() Item {
```